src/structure/structure.ts

```diff
@@ -326,44 +358,40 @@ export class StructureViewer {
      */
     public set active(guid: string) {
         const old = this.active;
-        assert(this._viewers.has(guid));
-        assert(this._viewers.has(old));
-
-        this._active = guid;
-
-        // change tooltip text in the active marker
-        const oldButton = getByID(`chsp-activate-${old}`);
-        oldButton.classList.toggle('chsp-active-structure', false);
-        const oldTooltip = oldButton.parentElement!.querySelector('.chsp-tooltip');
-        assert(oldTooltip !== null);
-        oldTooltip.innerHTML = 'choose as active';
-
-        const newButton = getByID(`chsp-activate-${this.active}`);
-        newButton.classList.toggle('chsp-active-structure', true);
-        const tooltip = newButton.parentElement!.querySelector('.chsp-tooltip');
-        assert(tooltip !== null);
-        tooltip.innerHTML = 'this is the active viewer';
-
-        // change style of the cell border
-        const oldCell = getByID(`gi-${old}`);
-        oldCell.classList.toggle('chsp-structure-viewer-cell-active', false);
-
-        const newCell = getByID(`gi-${this.active}`);
-        newCell.classList.toggle('chsp-structure-viewer-cell-active', true);
-
-        this._delay = getByID<HTMLInputElement>(`chsp-${this.active}-playback-delay`);
 
-        const data = this._viewers.get(this.active);
-        assert(data !== undefined);
-        let indexes;
-        if (this._indexer.mode === 'structure') {
-            indexes = this._indexer.from_structure_atom(data.current.structure);
-        } else {
-            const structure = data.current.structure;
-            const atom = data.current.atom;
-            indexes = this._indexer.from_structure_atom(structure, atom);
+        // choose *if* over assertion -- there's a weird error that I cannot pin-
+        // point where active is being passed an undefined.
```

Comment:
I'll try to give this a look, it might come from the constructor calling `this.active = '...'` before filling in `this.widgets`